webplugin/js/app/components/answer-feedback-service.js

```diff
@@ -0,0 +1,210 @@
+class AnswerFeedback {
+    constructor(options) {
+        /**
+         * end user widget configuration options.
+         * @type {Object.<string, Object>}
+         */
+
+        this.options = options;
+
+        /**
+         * Map to store messages for traversing to the end user message.
+         * @type {Object.<string, {
+         *   msg: Object,
+         *   floatWhere: string,
+         *   previousSibling: (Object|null)
+         * }>}
+         */
+
+        this.msgMap = {};
+
+        /**
+         * Array to maintain the order of message keys.
+         * @type {Array.<string>}
+         */
+        this.msgKeyMap = [];
+    }
+
+    /**
+     * @param {Object} data - The message data to set.
+     * @param {Object} data.msg - The message object.
+     * @param {string} data.msg.key - The unique key for the message.
+     * @returns {void}
+     */
+
+    set msgMapData(data) {
+        if (!CURRENT_GROUP_DATA.isConversationAssigneeBot) {
+            return;
+        }
+        const msgKey = data.msg.key;
+        const alreadyPresent = this.msgMap[msgKey];
+
+        this.msgMap[msgKey] = data;
+
+        // msg Key will be not changed that's why we are not updating the msgKeyMap
+        !alreadyPresent && this.msgKeyMap.push(msgKey);
+    }
+
+    handleOnFeedbackClick = (feedback, data) => {
+        const { payload, method = 'POST' } = this.getPayload(feedback, data);
+
+        mckUtils.ajax({
+            type: method,
+            url: `${Kommunicate.getBaseUrl()}/rest/ws/answer-feedback`,
+            headers: {
+                'x-authorization': window.Applozic.ALApiService.AUTH_TOKEN,
+            },
+            data: JSON.stringify({ payload }),
+            global: false,
+            contentType: 'application/json',
+            success: (result) => {
+                if (result.code == 'SUCCESS') {
+                    const msgMeta =
+                        this.msgMap[data.msg.key]?.msg?.metadata || {};
+                    msgMeta.KM_ANSWER_FEEDBACK = feedback;
+                } else {
+                    console.error('Failed to submit feedback');
+                }
+            },
+            error: function (data) {
+                console.error(data);
+            },
+        });
+    };
+
+    getPayload = (feedback, data) => {
+        const {
+            msg: { metadata, key },
+            assigneeKey,
+        } = data;
+
+        const answer = this.msgMap[key];
+        let question = answer;
+
+        // find the user message
+        while (
+            !metadata.KM_CONTEXT_QUESTION &&
+            question?.floatWhere !== 'mck-msg-right'
```

Comment:
this will be based on RTL setting?

Reply:
we are changing the class if it's rtl or ltr. This will work